freeipa.spec.in

```diff
@@ -400,47 +312,15 @@ and integration with Active Directory based infrastructures (Trusts).
 If you are installing an IPA server, you need to install this package.
 
 
-%if 0%{?with_python2}
-
-%package -n python2-ipaserver
-Summary: Python libraries used by IPA server
-Group: System Environment/Libraries
-BuildArch: noarch
-%{?python_provide:%python_provide python2-ipaserver}
-%{!?python_provide:Provides: python-ipaserver = %{version}-%{release}}
-Requires: %{name}-server-common = %{version}-%{release}
-Requires: %{name}-common = %{version}-%{release}
-Requires: python2-augeas
-Requires: python2-custodia >= 0.3.1
-Requires: python2-dbus
-Requires: python2-dns >= 1.15
-Requires: python2-gssapi >= 1.2.0-5
-Requires: python2-ipaclient = %{version}-%{release}
-Requires: python2-kdcproxy >= 0.3
-Requires: python2-ldap >= %{python_ldap_version}
-Requires: python2-lxml
-Requires: python2-pki >= %{pki_version}
-Requires: python2-pyasn1 >= 0.3.2-2
-Requires: python2-sssdconfig
-Requires: rpm-libs
-
-%description -n python2-ipaserver
-IPA is an integrated solution to provide centrally managed Identity (users,
-hosts, services), Authentication (SSO, 2FA), and Authorization
-(host access control, SELinux user roles, services). The solution provides
-features for further integration with Linux based clients (SUDO, automount)
-and integration with Active Directory based infrastructures (Trusts).
-If you are installing an IPA server, you need to install this package.
-
-%endif  # with_python2
-
 %package -n python3-ipaserver
 Summary: Python libraries used by IPA server
 Group: System Environment/Libraries
 BuildArch: noarch
 %{?python_provide:%python_provide python3-ipaserver}
 Requires: %{name}-server-common = %{version}-%{release}
 Requires: %{name}-common = %{version}-%{release}
+Conflicts: python2-ipaserver
+Obsoletes: python2-ipaserver < %{version}
```

Comment:
I forget the rpm behavior here but with a Conflicts isn't this going to prevent upgrades? Should it have a provides python2-ipaserver instead?

Upgrades are indeed not working:
# rpm -Uvh *
error: Failed dependencies:
        freeipa-common = 4.6.90.pre2-3.fc28 is needed by (installed) python2-ipalib-4.6.90.pre2-3.fc28.noarch

Reply:
Too bad. :-1: Then we need another way to replace and remove the old packages, then. Provides sounds wrong. We are not providing python2 packages.

Reply:
The specfile was missing a conflict and obsoletes for ``python2-ipalib``. Let's see that does the trick.